src/main/java/uk/gov/hmcts/reform/lrdapi/service/impl/RegionServiceImpl.java

```diff
@@ -18,13 +39,97 @@
     @Autowired
     RegionRepository regionRepository;
 
-    public LrdRegionResponse retrieveRegionByRegionDescription(String description) {
-        Region region = regionRepository.findByDescriptionIgnoreCase(description);
+    @Value("${loggingComponentName}")
+    private String loggingComponentName;
+
+    public List<LrdRegionResponse> retrieveRegionDetails(String regionId, String description) {
+        if (isNotBlank(regionId)) {
+            return retrieveRegionByRegionId(regionId);
+        }
```

Comment:
I think we can create single Blank method and use for both. Nothing harm in current code but it will reduce no of lines :)